Refactor split helper documentation for clarity

cathysarisky opened the pull request:
Updated the usage syntax for the split helper from `{{split}}` to `{{#split}}` and added additional examples for clarity, clarifying the behavior for empty strings.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates `split` helper docs to use `{{#split}}`, adds inline usage example, and documents filtering of empty strings.
> 
> - **Docs: `themes/helpers/utility/split.mdx`**
>   - Update usage and references from `{{split}}` to `{{#split}}`.
>   - Add **Inline mode** example using `{{#foreach (split ...)}}`.
>   - Document behavior: filters out empty strings when separators are sequential, with example output.
>   - Expand examples for custom separators and iteration with `{{#foreach}}`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e4b126976d39235cc0a239fc524f5f462c1cf420. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

JohnONolan commented:
Deploy preview for *tryghost* ready!

Built with commit 5a9adc135d52548994cd07a4b3ff661c3850f6ea

https://deploy-preview-22--tryghost.netlify.com

gargol commented:
oh nice! :heart: this way of deploying

cathysarisky committed:
Refactor split helper documentation for clarity

Updated the usage syntax for the split helper from `{{split}}` to `{{#split}}` and added additional examples for clarity, clarifying the behavior for empty strings.
